### PR TITLE
Make ocamlobjinfo work for Flambda 2 .cmx files

### PR DESCRIPTION
**Pull request by mshinwell**
Straightforward.

### CODE REVIEW

File: tools/objinfo.ml [hunk 1]
```diff
@@ -167,8 +167,12 @@ let print_cmx_infos (ui, crc) =
     if not !no_code then
       Format.printf "functions@ %a@.@."
         Export_info.print_functions export
-  | Flambda2 _ ->
-    failwith "Flambda2 objinfo support not yet merged"
+  | Flambda2 None ->
+    printf "This is an Flambda 2 object with no export info\n"
+  | Flambda2 (Some cmx) ->
+    printf "This is an Flambda 2 object\n";
```
Comment (xclerc) [inline tools/objinfo.ml:173]:
```suggestion
    printf "Flambda 2 export information:\n"
```

File: tools/objinfo.ml [hunk 2]
```diff
@@ -167,8 +167,12 @@ let print_cmx_infos (ui, crc) =
     if not !no_code then
       Format.printf "functions@ %a@.@."
         Export_info.print_functions export
-  | Flambda2 _ ->
-    failwith "Flambda2 objinfo support not yet merged"
+  | Flambda2 None ->
+    printf "This is an Flambda 2 object with no export info\n"
```
Comment (xclerc) [inline tools/objinfo.ml:171]:
```suggestion
    printf "Flambda 2 unit\n"
```

**Review by xclerc — APPROVED** [review]
Looks fine, for the sake of consistency we might want to
change the format strings to the `printf` calls
(See suggestions).

File: tools/objinfo.ml (hunk 1 above)
Comment (mshinwell) [reply tools/objinfo.ml]: Fixed

File: tools/objinfo.ml (hunk 2 above)
Comment (mshinwell) [reply tools/objinfo.ml]: I've fixed a modified version of this, I'd like to see explicitly that it doesn't have export info, to avoid confusion.

**Comment by xclerc** [comment]
(Re-approving after `fe5ac06`.)

**Pull request merged**